components/content_settings/core/browser/brave_content_settings_pref_provider_unittest.cc

```diff
@@ -239,10 +243,59 @@ class ShieldsFingerprintingSetting : public ShieldsSetting {
 
 class ShieldsHTTPSESetting : public ShieldsSetting {
  public:
-  explicit ShieldsHTTPSESetting(BravePrefProvider* provider)
+  explicit ShieldsHTTPSESetting(BravePrefProvider* provider,
+                                DefaultProvider* default_provider)
       : ShieldsSetting(
             provider,
-            {{GURL(), ContentSettingsType::BRAVE_HTTP_UPGRADABLE_RESOURCES}}) {}
+            {{GURL(), ContentSettingsType::BRAVE_HTTP_UPGRADABLE_RESOURCES}}),
+        default_provider_(default_provider) {}
+
+  void SetPreMigrationDefaultSetting(ContentSetting setting) {
+    GURL allowed_url("https://allowed.com");
+    GURL blocked_url("https://blocked.com");
+    GURL default_url("https://default.com");
+
+    SetPreMigrationSettings(ContentSettingsPattern::FromURL(allowed_url),
+                            CONTENT_SETTING_ALLOW);
+    CheckSettingsWouldAllow(allowed_url);
+    SetPreMigrationSettings(ContentSettingsPattern::FromURL(blocked_url),
+                            CONTENT_SETTING_BLOCK);
+    CheckSettingsWouldBlock(blocked_url);
+
+    provider_->GetPref(ContentSettingsType::BRAVE_HTTP_UPGRADABLE_RESOURCES)
```

Comment:
you need to add an old-style wildcard content setting and check that it gets migrated

Reply:
this is what actially this test does, this lines adding old-style prefs.